OpenRA.Mods.Common/Activities/Air/FlyOffMap.cs

```diff
@@ -32,9 +34,18 @@ public override Activity Tick(Actor self)
 				return NextActivity;
 			}
 
-			if (IsCanceled || !self.World.Map.Contains(self.Location))
+			if (IsCanceled)
 				return NextActivity;
 
+			if (!self.World.Map.Contains(self.Location))
+			{
+				if (removeOnMapLeft)
+					Queue(new RemoveSelf());
```

Comment:
Lowering the risk that a coder forgets to queue `RemoveSelf`. `ParadropProperties` has that bug on bleed, in fact.

I can of course queue `RemoveSelf` there instead and leave this activity as-is if you prefer that, but I felt it would be more future-proof if `FlyOffMap` did that by default, as I can't think of anything realistic where we may want an actor to leave the map but not get removed.

Reply:
My concern here is that this introduces a new one-off code pattern for queuing an activity, and history shows that this usually doesn't end well.

We can get the same result with less complexity by folding the removal directly into the activity:
```csharp
public override Activity Tick(Actor self)
{
	// Refuse to take off if it would land immediately again.
	if (aircraft.ForceLanding)
 	{
 		Cancel(self);
		return NextActivity;
	}

	if (IsCanceled)
		return NextActivity;

	// Fly towards the edge of the map then dispose the actor once it is outside
	// TODO: We will want a minimum margin here so that the visible bounds of the aircraft are fully outside the map
	// TODO: Make sure that this works for CanHover/VTOL aircraft after activities have been merged
	if (self.World.Map.Contains(self.Location))
		Fly.FlyToward(self, aircraft, aircraft.Facing, aircraft.Info.CruiseAltitude);
	else
		self.Dispose();

	return this;
}
```